content/en/docs/concepts/architecture/_index.md

```diff
@@ -106,6 +106,29 @@ kube-proxy on the nodes in your cluster.
 
 {{< glossary_definition term_id="container-runtime" length="all" >}}
 
+## Ability to self-heal
+
+The idea behind self-healing Kubernetes is simple: If a container fails, Kubernetes automatically redeploys the afflicted 
+container to its desired state to restore operations
+
+Self-healing Kubernetes has four capabilities :
+
+ - restart failed containers.
+ - replace containers that require updates, such as a new software version.
+ - disable containers that don't respond to predefined health checks.
+ - prevent containers from appearing to users or other containers until they are ready.
+
```

Comment:
- Restart failed containers.
- Replace containers that require updates, such as a new software version.
- Kill containers that don't respond to user-defined health checks.
- Prevent advertising containers to clients until they are ready to serve traffic.